packages/@aws-cdk/aws-cloudfront/lib/distribution.ts

```diff
@@ -1,14 +1,332 @@
+      cacheBehaviors: Lazy.anyValue({ produce: () => this.renderCacheBehaviors() }),
+      viewerCertificate: this.certificate ? { acmCertificateArn: this.certificate.certificateArn } : undefined,
+      customErrorResponses: this.renderCustomErrorResponses(),
+      priceClass: props.priceClass ?? undefined,
+    } });
+
+    this.domainName = distribution.attrDomainName;
+    this.distributionId = distribution.ref;
+  }
+
+  /**
+   * Adds a new behavior to this distribution for the given pathPattern.
+   *
+   * @param pathPattern the path pattern (e.g., 'images/*') that specifies which requests to apply the behavior to.
+   * @param behaviorOptions the options for the behavior at this path.
+   */
+  public addBehavior(pathPattern: string, origin: Origin, behaviorOptions: AddBehaviorOptions = {}) {
+    if (pathPattern === '*') {
+      throw new Error('Only the default behavior can have a path pattern of \'*\'');
+    }
+    this.additionalBehaviors.push(new CacheBehavior({ pathPattern, origin, ...behaviorOptions }));
+    this.addOrigin(origin);
+  }
+
+  private addOrigin(origin: Origin) {
+    if (!this.origins.has(origin)) {
+      this.origins.add(origin);
+      origin.bind(this, { originIndex: this.origins.size });
+    }
+  }
+
+  private renderOrigins(): CfnDistribution.OriginProperty[] {
+    const renderedOrigins: CfnDistribution.OriginProperty[] = [];
+    this.origins.forEach(origin => renderedOrigins.push(origin._renderOrigin()));
+    return renderedOrigins;
+  }
+
+  private renderCacheBehaviors(): CfnDistribution.CacheBehaviorProperty[] | undefined {
+    if (this.additionalBehaviors.length === 0) { return undefined; }
+    return this.additionalBehaviors.map(behavior => behavior._renderBehavior());
+  }
+
+  private renderCustomErrorResponses(): CfnDistribution.CustomErrorResponseProperty[] | undefined {
+    if (this.errorConfigurations.length === 0) { return undefined; }
+    function validateCustomErrorResponse(errorResponse: CustomErrorResponse) {
+      if (errorResponse.responsePagePath && !errorResponse.responseCode) {
+        throw new Error('\'responseCode\' must be provided if \'responsePagePath\' is defined');
+      }
+      if (!errorResponse.responseCode && !errorResponse.errorCachingMinTtl) {
+        throw new Error('A custom error response without either a \'responseCode\' or \'errorCachingMinTtl\' is not valid.');
+      }
+    }
+    this.errorConfigurations.forEach(e => validateCustomErrorResponse(e));
+
+    return this.errorConfigurations.map(errorConfig => {
+      return {
+        errorCachingMinTtl: errorConfig.errorCachingMinTtl?.toSeconds(),
+        errorCode: errorConfig.errorCode,
+        responseCode: errorConfig.responseCode,
+        responsePagePath: errorConfig.responsePagePath,
+      };
+    });
+  }
+
+}
+
+/**
+ * The price class determines how many edge locations CloudFront will use for your distribution.
+ */
+export enum PriceClass {
+  PRICE_CLASS_100 = 'PriceClass_100',
```

Comment:
missing docstrings on enum members (copy from AWS docs)

Reply:
ACK. This is moved from the existing code; will add docstrings. FWIW, the only real documentation for the price classes is in a table that's a bit hard to distill down. That being said, I'll give it a shot.